File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/ui/mypage/CharacterEditUiState.kt
package org.sesacteamproject.passmate.ui.mypage

data class CharacterEditUiState(
    val avatarId: Int? = null,
    val isLoading: Boolean = true,
    val hasLoadError: Boolean = false,
    // 제출 in-flight — 중복 호출 방지 (규칙 §9)
    val isSubmitting: Boolean = false
) {

    val canSubmit: Boolean
        get() = !isLoading && !isSubmitting && avatarId != null
}

File: iosApp/iosApp/ui/mypage/CharacterEditUiState.swift
// Compose CharacterEditUiState.kt 미러 (M-12-7)
struct CharacterEditUiState {
    var avatarId: Int?

    var isLoading: Bool = true

    var hasLoadError: Bool = false

    // 제출 in-flight — 중복 호출 방지 (규칙 §9)
    var isSubmitting: Bool = false

    var canSubmit: Bool {
        !isLoading && !isSubmitting && avatarId != nil
    }
}
